Bookish/Controllers/BookController.cs

```diff
@@ -152,8 +177,45 @@ public async Task<ActionResult> BookInput(BookInput input)
         }
         return View();
     }
+
+    [HttpGet]
+    public async Task<ActionResult> BookEdited(Book input)
+    {
+        string bookId = input.Id;
+        using (var context = new EFCore())
+        {
+            var book = context.Books.Find(bookId);
+            if (book != null)
+            {
+                if (input.Title != null) book.Title = input.Title;
+                if (input.Author != null) book.Author = input.Author;
+                context.SaveChanges();
+                return View();
+            }
+
+            return View("BookNotEdited");
+        }
+    }
     
-    
+    [HttpGet]
+    public async Task<ActionResult> BookDeleted(Book input)
+    {
+        string bookId = input.Id;
+        using (var context = new EFCore())
+        {
+            var book = context.Books.Find(bookId);
+            if (book != null)
+            {
+                context.Books.Remove(book);
+                context.SaveChanges();
+                return View();
+            }
+
+            return View("BookNotEdited");
+        }
+    }
+
+
     public async Task<ActionResult> Catalogue()
```

Comment:
What kind of method is this? GET? POST?

Reply:
Also, why is it a `Task<ActionResult>` and not just an `ActionResult`? `Task` implies you have asynchronous code, but it doesn't look like you do. This applies for most of the other methods in this class too. If you aren't `await`ing anything, you don't have async code